scene/gui/foldable_container.cpp

```diff
@@ -0,0 +1,657 @@
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "text_direction", PROPERTY_HINT_ENUM, "Auto,Left-to-Right,Right-to-Left,Inherited"), "set_text_direction", "get_text_direction");
+	ADD_PROPERTY(PropertyInfo(Variant::STRING, "language", PROPERTY_HINT_LOCALE_ID), "set_language", "get_language");
+
+	BIND_ENUM_CONSTANT(POSITION_TOP);
+	BIND_ENUM_CONSTANT(POSITION_BOTTOM);
+
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_STYLEBOX, FoldableContainer, title_style, "title_panel");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_STYLEBOX, FoldableContainer, title_hover_style, "title_hover_panel");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_STYLEBOX, FoldableContainer, title_collapsed_style, "title_collapsed_panel");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_STYLEBOX, FoldableContainer, title_collapsed_hover_style, "title_collapsed_hover_panel");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_STYLEBOX, FoldableContainer, focus_style, "focus");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_STYLEBOX, FoldableContainer, panel_style, "panel");
+
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_FONT, FoldableContainer, title_font, "font");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_FONT_SIZE, FoldableContainer, title_font_size, "font_size");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_CONSTANT, FoldableContainer, title_font_outline_size, "outline_size");
+
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_COLOR, FoldableContainer, title_font_color, "font_color");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_COLOR, FoldableContainer, title_hovered_font_color, "hover_font_color");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_COLOR, FoldableContainer, title_collapsed_font_color, "collapsed_font_color");
+	BIND_THEME_ITEM_CUSTOM(Theme::DATA_TYPE_COLOR, FoldableContainer, title_font_outline_color, "font_outline_color");
+
+	BIND_THEME_ITEM(Theme::DATA_TYPE_ICON, FoldableContainer, expanded_arrow);
+	BIND_THEME_ITEM(Theme::DATA_TYPE_ICON, FoldableContainer, expanded_arrow_mirrored);
+	BIND_THEME_ITEM(Theme::DATA_TYPE_ICON, FoldableContainer, folded_arrow);
+	BIND_THEME_ITEM(Theme::DATA_TYPE_ICON, FoldableContainer, folded_arrow_mirrored);
+
+	BIND_THEME_ITEM(Theme::DATA_TYPE_CONSTANT, FoldableContainer, h_separation);
+}
+
+FoldableContainer::FoldableContainer(const String &p_text) {
+	text_buf.instantiate();
+	set_text(p_text);
+	set_focus_mode(FOCUS_ALL);
+	set_mouse_filter(MOUSE_FILTER_STOP);
+}
+
+FoldableContainer::~FoldableContainer() {
+	if (foldable_group.is_valid()) {
+		foldable_group->containers.erase(this);
+	}
+}
+
+FoldableContainer *FoldableGroup::get_expanded_container() const {
+	for (FoldableContainer *container : containers) {
+		if (!container->is_folded()) {
+			return container;
+		}
+	}
+
+	return nullptr;
+}
+
+void FoldableGroup::set_allow_folding_all(bool p_enabled) {
+	allow_folding_all = p_enabled;
+	if (!allow_folding_all && !get_expanded_container() && containers.size() > 0) {
+		updating_group = true;
+		for (FoldableContainer *E : containers) {
+			E->set_folded(false);
+			break;
```

Comment:
```suggestion
		for (FoldableContainer *container : containers) {
			container->set_folded(false);
			break;
```

Reply:
Also this section fails to compile on MSVC for me because it thinks this is "unreachable code"? It's clearly a false positive, but I couldn't tell you where it's being tripped up

Reply:
Maybe it complains that the loop always ends at the first iteration?
Not sure how to pick an element from HashSet otherwise.

Reply:
If you're just getting the first iteration, wouldn't `begin()` do the trick?

Reply:
Fixed.